app/Models/Question.php

```diff
@@ -36,4 +37,24 @@ public function choices(): HasMany
     {
         return $this->hasMany(Choice::class);
     }
+
+    public function getChoices()
+    {
+        return Options::forModels(
+            Choice::query()->whereIn('id', $this->choices->modelKeys()),
+            label: fn (Choice $choice) => $choice->getTranslation('label', locale())
+        )->toArray();
+    }
+
+    public function getCorrectChoices()
+    {
+        $correctChoices = [];
+        foreach ($this->choices as $choice) {
+            if ($choice->is_answer) {
+                $correctChoices[] = $choice->id;
+            }
+        }
+
+        return $correctChoices;
```

Comment:
It's been a while so I don't quite remember clearly, but while working on this task, Ned and I realized that it would be more convenient for us to have more granular model designs. Is there better design you want to explore?

Reply:
Well that's the question I was hoping to figure out. I'm trying to understand the separation of concerns here and wondering if there were future plans or access requirements for the structure. For example, would we want to reuse questions or choices and make them many-to-many relationships. One thing I'm sort of finding a bit strange is that the question doesn't know the correct answer, but the choice knows if it is correct.

For example if we had a questions like "Who should you contact if you have further concerns?". Suddenly the answer to the question depends on the context (i.e the quiz). The correct choice depends on the question which depends on the quiz.

Now let's say that we will never re-use questions and choices, so they are always a one-to-one relationship. Can we just make them an array of options on the question model? Is it for localization that they are separate models? Should the question really be the one that knows the correct answer?

Reply:
To summarize from the discussion we had yesterday at the IDRC planning meeting:

- General suggestion to make the system more flexible, especially because we don't really know the full requirements
- Allow for quizzes at the course and/or module level
  - This will mean that we need to determine when the modules are complete by checking if the the videos have finished and if the available module quizzes have been passed.
  - The course is complete if all modules are complete and if there is a course quiz that it has passed
- The questions should be available to be added to any quiz, many-to-many relationship 
- The choices will be part of a question as opposed to being in a separate table. There will be an array of choices, and an array of correct answers which is an array of the indexes of the choices that are correct. Both of these arrays are required.
  - Need to make sure that we can localize the choices.